crates/uv-resolver/src/lock/mod.rs

```diff
@@ -1786,6 +1786,16 @@ impl Package {
                 .collect::<Result<_, _>>()
                 .map_err(LockErrorKind::RequirementRelativePath)?
         };
+        let provides_extras = if id.source.is_immutable() {
+            Vec::new()
```

Comment:
Something we need to consider: if we want to start using this to _enforce_ that a given extra exists (at install time, by reading from the lockfile), we won't be able to tell the difference between "empty" and "absent". So we either need to _not_ do that, or bump the lockfile version so we can tell the difference, or write these even when empty.

Reply:
I'm not sure. @zanieb, do you have an opinion? Bumping the lockfile version means that older uv versions will reject newer lockfiles (but we can still read older lockfiles in newer uv versions).

Reply:
In addition to always writing this we'd also need to make this like... an `Option<Vec<ExtraName>>` to distinguish whether we found it in the lockfile, right?

Reply:
Yeah that’s right

Reply:
Added the commit, I'm not sure we're gonna love how effectively `[package.metadata]` is no longer ever empty-and-therefore-omitable.

Reply:
I was thinking that we'd continue to omit it for immutable sources, is that doable? It should massively reduce the diff.

Reply:
It did remove a lot of them, yeah. (updated the commit)

Reply:
Generally in favor of less breaking lockfile changes :) it's decent timing with 0.6 coming up, but from what I understand this doesn't justify the churn?

Reply:
It's just to enable more validation on cli arguments, yeah. Not really worth a break.

Reply:
Yeah I don't think we should bump for it. I do wonder if we should treat `requires-dist` the same way, though: always write it, even if empty?

I think users will likely be annoyed that the lockfile is churning more (even without a bump) in that you're now getting different results if you lock with a later version (even though it should be backwards-compatible), but I don't know how we avoid that.